gammapy/estimators/utils.py

```diff
@@ -778,71 +785,130 @@ def combine_flux_maps(maps, method="gaussian_errors", reference_model=None):
     flux_maps : `~gammapy.estimators.FluxMaps`
         Joint flux map.
     """
-    if method != "gaussian_errors":
-        raise ValueError(
-            f'Invalid method : {method}, available methods are : "gaussian_errors"'
-        )
 
     if isinstance(maps, FluxMaps):
-        geom = maps.dnde.sum_over_axes().geom
-        means = list(maps.dnde.copy().iter_by_image(keepdims=True))
-        sigmas = list(maps.dnde_err.copy().iter_by_image(keepdims=True))
-        mean = Map.from_geom(geom, data=means[0].data, unit=means[0].unit)
-        sigma = Map.from_geom(geom, data=sigmas[0].data, unit=sigmas[0].unit)
         gti = maps.gti
         meta = maps.meta
         if reference_model is None:
             reference_model = maps.reference_model
     else:
-        means = [map_.dnde.copy() for map_ in maps]
-        sigmas = [map_.dnde_err.copy() for map_ in maps]
-        mean = means[0]
-        sigma = sigmas[0]
-
         gtis = [map_.gti for map_ in maps if map_.gti is not None]
         if np.any(gtis):
             gti = gtis[0].copy()
             for k in range(1, len(gtis)):
                 gti.stack(gtis[k])
         else:
             gti = None
-
-        if reference_model is None:
-            reference_model = maps[0].reference_model
-
         # TODO : change this once we have stackable metadata objets
         metas = [map_.meta for map_ in maps if map_.meta is not None]
         meta = {}
         if np.any(metas):
             for data in metas:
                 meta.update(data)
+        if reference_model is None:
+            reference_model = maps[0].reference_model
+
+    if method == "gaussian_errors":
+
+        if isinstance(maps, FluxMaps):
+            geom = maps.dnde.sum_over_axes().geom
+            means = list(maps.dnde.copy().iter_by_image(keepdims=True))
+            sigmas = list(maps.dnde_err.copy().iter_by_image(keepdims=True))
+            mean = Map.from_geom(geom, data=means[0].data, unit=means[0].unit)
+            sigma = Map.from_geom(geom, data=sigmas[0].data, unit=sigmas[0].unit)
+
+        else:
```

Comment:
In this case, one should have a list of FluxMaps, status that you do not test explicitly. Is it needed?